workflow/rules/benchmarking.smk

```diff
@@ -206,6 +206,271 @@ rule plot_strain_call_error:
         "../scripts/plot-caller-error.py"
 
 
+rule assembly_comparison_megahit_std:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs="results/{date}/assembly/{sample}/megahit-std/{sample}.contigs.fasta",
+    log:
+        "logs/{date}/megahit-std/{sample}.log",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    threads: 8
+    conda:
+        "../envs/megahit.yaml"
+    shell:
+        "(megahit -1 {input.fastq1} -2 {input.fastq2} --out-dir {params.outdir} -f && "
+        "mv {params.outdir}/final.contigs.fa {output.contigs} ) > {log} 2>&1"
+
+
+rule assembly_comparison_megahit_meta_large:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs=(
+            "results/{date}/assembly/{sample}/megahit-meta-large/{sample}.contigs.fasta"
+        ),
+    log:
+        "logs/{date}/megahit-meta-large/{sample}.log",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    threads: 8
+    conda:
+        "../envs/megahit.yaml"
+    shell:
+        "(megahit -1 {input.fastq1} -2 {input.fastq2} --presets meta-large --out-dir {params.outdir} -f && "
+        "mv {params.outdir}/final.contigs.fa {output.contigs} ) > {log} 2>&1"
+
+
+rule assembly_comparison_megahit_meta_sensitive:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs="results/{date}/assembly/{sample}/megahit-meta-sensitive/{sample}.contigs.fasta",
+    log:
+        "logs/{date}/megahit-meta-sensitive/{sample}.log",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    threads: 8
+    conda:
+        "../envs/megahit.yaml"
+    shell:
+        "(megahit -1 {input.fastq1} -2 {input.fastq2} --presets meta-sensitive --out-dir {params.outdir} -f && "
+        "mv {params.outdir}/final.contigs.fa {output.contigs} ) > {log} 2>&1"
```

Comment:
All the megahit rules are basically the same. Just have one, add a wildcard for the preset (e.g. `{megahit_preset}` and set the parameters accordingly.

Something like this:
```suggestion
rule assembly_comparison_megahit_meta_sensitive:
    input:
        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
    output:
        contigs="results/{date}/assembly/{sample}/megahit-{preset}/{sample}.contigs.fasta",
    log:
        "logs/{date}/megahit-meta-sensitive/{sample}.log",
    params:
        outdir=lambda w, output: os.path.dirname(output[0]),
    threads: 8
    conda:
        "../envs/megahit.yaml"
    shell:
        "(megahit -1 {input.fastq1} -2 {input.fastq2} --presets {wildcards.preset} --out-dir {params.outdir} -f && "
        "mv {params.outdir}/final.contigs.fa {output.contigs} ) > {log} 2>&1"
```